lib/segment/src/vector_storage/tests/test_appendable_multi_dense_vector_storage.rs

```diff
@@ -217,62 +226,65 @@ fn do_test_update_from_delete_points(storage: Arc<AtomicRefCell<VectorStorageEnu
+    storage_type: MultiDenseStorageType,
+    path: &Path,
+) -> Arc<AtomicRefCell<VectorStorageEnum>> {
+    match storage_type {
+        MultiDenseStorageType::SimpleRamFloat => {
+            let db = open_db(path, &[DB_VECTOR_CF]).unwrap();
+            open_simple_multi_dense_vector_storage(
+                db,
+                DB_VECTOR_CF,
+                4,
+                Distance::Dot,
+                MultiVectorConfig::default(),
+                &AtomicBool::new(false),
+            )
+            .unwrap()
+        }
+        MultiDenseStorageType::AppendableMmapFloat => open_appendable_memmap_multi_vector_storage(
+            path,
             4,
             Distance::Dot,
             MultiVectorConfig::default(),
             &AtomicBool::new(false),
         )
-        .unwrap();
+        .unwrap(),
+    }
+}
+
+#[rstest]
+fn test_delete_points_in_simple_multi_dense_vector_storage(
+    #[values(
+        MultiDenseStorageType::SimpleRamFloat,
+        MultiDenseStorageType::AppendableMmapFloat
+    )]
+    storage_type: MultiDenseStorageType,
+) {
+    let dir = Builder::new().prefix("storage_dir").tempdir().unwrap();
+
+    {
+        let storage = create_vector_storage(storage_type, dir.path());
         do_test_delete_points(storage.clone());
         storage.borrow().flusher()().unwrap();
     }
-    let db = open_db(dir.path(), &[DB_VECTOR_CF]).unwrap();
-    let _storage = open_simple_multi_dense_vector_storage(
-        db,
-        DB_VECTOR_CF,
-        4,
-        Distance::Dot,
-        MultiVectorConfig::default(),
-        &AtomicBool::new(false),
-    )
-    .unwrap();
+    let _storage = create_vector_storage(storage_type, dir.path());
 }
 
-#[test]
-fn test_update_from_delete_points_simple_multi_dense_vector_storage() {
+#[rstest]
+fn test_update_from_delete_points_simple_multi_dense_vector_storage(
+    #[values(
+        MultiDenseStorageType::SimpleRamFloat,
+        MultiDenseStorageType::AppendableMmapFloat
+    )]
+    storage_type: MultiDenseStorageType,
+) {
     let dir = Builder::new().prefix("storage_dir").tempdir().unwrap();
     {
-        let db = open_db(dir.path(), &[DB_VECTOR_CF]).unwrap();
-        let storage = open_simple_multi_dense_vector_storage(
-            db,
-            DB_VECTOR_CF,
-            4,
-            Distance::Dot,
-            MultiVectorConfig::default(),
-            &AtomicBool::new(false),
-        )
-        .unwrap();
+        let storage = create_vector_storage(storage_type, dir.path());
         do_test_update_from_delete_points(storage.clone());
         storage.borrow().flusher()().unwrap();
     }
-
-    let db = open_db(dir.path(), &[DB_VECTOR_CF]).unwrap();
-    let _storage = open_simple_multi_dense_vector_storage(
-        db,
-        DB_VECTOR_CF,
-        4,
-        Distance::Dot,
-        MultiVectorConfig::default(),
-        &AtomicBool::new(false),
-    )
-    .unwrap();
+    let _storage = create_vector_storage(storage_type, dir.path());
```

Comment:
can we retrieve the number of points before the drop and then retrieve after re-open to make sure that the persistence works